### PR TITLE
style: Move `float` CSS property outside mako

### LINKED ISSUE #21130: style: Move float outside mako
I like to take this one as sub issue of #19015 .

### PR DESCRIPTION
**Pull request by kleinph**
Hi, I am fairly new to Rust and this is my first contribution to a Rust project.

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21130

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

I'm not sure if the approach with the two enums is good. Please let me know if this can be improved. 
Regarding the tests I'm unsure: `tests/wpt/web-platform-tests/css/css-display/display-contents-float-001.html` seems to fail already before this PR, (and the cause is maybe not related to the style property definition).

Also I would like to add some docs, is it allowed to use the descriptions from MDN (https://developer.mozilla.org/en-US/docs/Web/CSS/float).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21139)

### CODE REVIEW

File: components/style/properties/longhands/box.mako.rs [hunk 1]
```diff
@@ -54,61 +54,15 @@ ${helpers.single_keyword("position", "static absolute relative fixed sticky",
...
-        fn from_computed_value(computed: &computed_value::T) -> SpecifiedValue {
-            match *computed {
-                % for value in "None Left Right".split():
-                    computed_value::T::${value} => SpecifiedValue::${value},
-                % endfor
-            }
-        }
-    }
-</%helpers:single_keyword>
+${helpers.predefined_type("float", "Float",
```
Comment (emilio) [inline components/style/properties/longhands/box.mako.rs]: Could we use block indentation for here?

```rust
${helpers.predefined_type(
    "float",
    "Float",
    "computed::Float::None",
    // ...
)}
```

**Review by emilio — APPROVED** [review]
It looks amazing, thank you!

Let me know if you don't have time for the indentation change, I can do it in a follow-up.

**Comment by emilio** [comment]
> I'm not sure if the approach with the two enums is good. Please let me know if this can be improved.
Regarding the tests I'm unsure: tests/wpt/web-platform-tests/css/css-display/display-contents-float-001.html seems to fail already before this PR, (and the cause is maybe not related to the style property definition).

The two enums is the right approach, given they have different variants. Tests should be unchanged, since this shouldn't introduce any behavior changes.

Pointing at the spec for this should be enough as docs.

File: components/style/properties/longhands/box.mako.rs (hunk 1 above)
Comment (kleinph) [reply components/style/properties/longhands/box.mako.rs]: Sure

**Comment by kleinph** [comment]
> It looks amazing, thank you!

Thanks! :)
> Let me know if you don't have time for the indentation change, I can do it in a follow-up.

No. I will update the PR today.

I also want to do `clear` in the same style. Can I add it to this PR (as a second commit) or in its own PR?

**Review by emilio — APPROVED** [review]
> I also want to do clear in the same style. Can I add it to this PR (as a second commit) or in its own PR?

I think its own PR would be better for tracking, and also would allow this one to merge sooner.

**Comment by emilio** [comment]
@bors-servo retry
 
 * Flaky WebSocket failures

**Comment by kleinph** [comment]
Sorry closed by accident

**Comment by emilio** [comment]
Hah, np!

@bors-servo r+

**Comment by jdm** [comment]
@bors-servo retry
* network problems

**Pull request merged**